src/components/hermes-skills/useSkillDetail.ts

```
@@ -84,10 +113,15 @@ export function useSkillDetail(inspectId: string | null, fromInstalled = false):
     if (!inspectId) return;
 
     const scope = fromInstalled ? '&scope=installed' : '';
-    const cacheKey = `${fromInstalled ? 'i' : 'b'}:${inspectId}`;
+    const cacheKey = detailKey(inspectId, fromInstalled);
     const controller = new AbortController();
     let cancelled = false;
 
+    // A run that is about to answer the question again owns the note: leaving
+    // the old one up left a failure sitting under a panel that had just loaded
+    // (refresh() after an install re-runs this effect with the same id).
+    setErrorState((held) => (held && held.key === cacheKey ? null : held));
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/hermes-skills/useSkillDetail.ts:123:5
  121 |     // the old one up left a failure sitting under a panel that had just loaded
  122 |     // (refresh() after an install re-runs this effect with the same id).
> 123 |     setErrorState((held) => (held && held.key === cacheKey ? null : held));
      |     ^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  124 |
  125 |     const cached = cache.current.get(cacheKey);
  126 |     if (cached) {

(react-hooks/set-state-in-effect)